jsonrpc/eth_endpoint.go

```diff
@@ -518,8 +527,11 @@ func (e *Eth) EstimateGas(arg *txnArgs, rawNum *BlockNumber) (interface{}, error
 	gasPriceInt := new(big.Int).Set(transaction.GasPrice)
 	valueInt := new(big.Int).Set(transaction.Value)
 
-	// If the sender address is present, recalculate the ceiling to his balance
-	if transaction.From != types.ZeroAddress && transaction.GasPrice != nil && gasPriceInt.BitLen() != 0 {
+	var availableBalance *big.Int
+
+	// If the sender address is present, figure out how much available funds
+	// are we working with
+	if transaction.From != types.ZeroAddress {
```

Comment:
Good catch, but couldn't we make this code more efficient by checking this as early as possible as well:

```
transaction.GasPrice != nil && gasPriceInt.BitLen() != 0) || transaction.Value // this means no reading acc balance necessary
```
?

Reply:
There's no point in checking if `transaction.GasPrice` or `transaction.Value` were set, as they will always have some value: 
* [transaction.Value](https://github.com/0xPolygon/polygon-edge/blob/41c34e15c6e1af945205b339736e7afa66a97873/jsonrpc/eth_endpoint.go#L958-L960)
* [transaction.GasPrice](https://github.com/0xPolygon/polygon-edge/blob/41c34e15c6e1af945205b339736e7afa66a97873/jsonrpc/eth_endpoint.go#L962-L964)

I'm not sure what you meant by having this check as early as possible?

Reply:
ping @lazartravica to resolve this communication